axelor-account/src/main/java/com/axelor/apps/account/web/PartnerController.java

```diff
@@ -55,4 +59,31 @@ public void getDefaultSpecificTaxNote(ActionRequest request, ActionResponse resp
     Partner partner = request.getContext().asType(Partner.class);
     response.setValue("specificTaxNote", partnerAccountService.getDefaultSpecificTaxNote(partner));
   }
+
+  public void openCustomerDebtRecovery(ActionRequest request, ActionResponse response) {
+    Partner partner = request.getContext().asType(Partner.class);
+    List<DebtRecovery> debtRecoveryList =
+        Beans.get(DebtRecoveryRepository.class)
+            .all()
+            .filter("self.accountingSituation.partner.id = ?1", partner.getId())
+            .fetch();
+    if (debtRecoveryList.size() == 1) {
+      response.setView(
+          ActionView.define("Debt Recovery")
+              .model(DebtRecovery.class.getName())
+              .add("form", "debt-recovery-form")
+              .add("grid", "debt-recovery-grid")
+              .context("_showRecord", debtRecoveryList.get(0).getId())
```

Comment:
Use '_showSingle' context variable instead of fetching records and checking size